porcupine/plugins/filemanager.py

```diff
@@ -150,6 +150,35 @@ def update_dialog_state(*junk: object) -> None:
     return new_path
 
 
+    def update_dialog_state(*junk: object) -> None:
+        if overwrite_var.get():
+            ok_button.config(state="normal")
+            entry.config(state="disabled")
+            return
+
+        entry.config(state="normal")
+        name = entry.get()
+        try:
+            possible_new_path = (target_dir / "dummy").with_name(name)
+        except ValueError:
+            ok_button.config(state="disabled")
+            return
+
+        ok_button["state"] = "disabled" if possible_new_path.exists() else "normal"
+
+    file_name_var.trace_add("write", update_dialog_state)
+    overwrite_var.trace_add("write", update_dialog_state)
+    entry.bind("<Return>", (lambda event: ok_button.invoke()), add=True)
+    entry.bind("<Escape>", (lambda event: cancel_button.invoke()), add=True)
+    entry.select_range(0, "end")
+    entry.focus()
+
+    dialog.title(dialog_title)
+    dialog.resizable(False, False)
+    dialog.wait_window()
+
+    return new_path
+
```

Comment:
You apparently forgot to delete some of the copy/paste. There's still two `return new_path`s, and the code after the first one never runs.